pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -648,7 +649,12 @@ func (w *Webhook) injectAutoInstruConfig(pod *corev1.Pod, libsToInject []libInfo
 				{
 					key:     dotnetProfilingLdPreloadKey,
 					valFunc: dotnetProfilingLdPreloadEnvValFunc,
-				}})
+					isEligibleToInject: func(c corev1.Container) bool {
+						idx := mutatecommon.EnvIndex(c.Env, "DD_PROFILING_ENABLED")
```

Comment:
So this is going to add the LD_PRELOAD variable as long as DD_PROFILING_ENABLED is set to a truthy value. But what if DD_PROFILING_ENABLED was set for a Java/Python/Ruby/JS process and not a .NET process? I think we'd run into the musl/glibc problem again for a non-dotnet process in an alpine container that enabled profiling.

Is there something else we can use to be sure that it's dotnet? Or should we just be pre-emptive and block it entirely for now?